interpreter/mesh/src/main/scala/io/buoyant/interpreter/mesh/Client.scala

```diff
@@ -334,9 +340,11 @@ object Client {
     case mesh.Endpoint(Some(_), Some(ipBuf), Some(port), pmeta) =>
       val ipBytes = Buf.ByteArray.Owned.extract(ipBuf)
       val ip = InetAddress.getByAddress(ipBytes)
-      val meta = Seq.empty[(String, Any)] ++
-        pmeta.flatMap(_.nodeName).map(Metadata.nodeName -> _)
-      Address.Inet(new InetSocketAddress(ip, port), Addr.Metadata(meta: _*))
+      Address
+        .Inet(
+          new InetSocketAddress(ip, port),
+          pmeta.mapValues(v => Try(mapper.readValue[Map[String, Any]](v)).getOrElse(v))
```

Comment:
If the protobuf filed is a `map<string, string>` then we don't need to use an objectMapper and deserialize JSON.

Reply:
My initial thought was that the proto field is a string that represents either a leaf json value or a json object, hence I was trying to deserialize it as a tree first, to avoid the effect of having a json object represented as a string. However, now that I see that the consul api enforces all values to be string in the metadata map, then this gymnastics is not needed at all.